### ISSUE TITLE
Suggestion for improvement. Add the ability to use flags for the batch manager.

### ISSUE THREAD

**Issue by xpamych** [Arsen6331/lure#53]
It is not possible to use the flags of the package manager, in cases where it is necessary to overwrite or ignore the package. For example, go-bin: in my distribution, the version is lower than in lure and it tries to install it when updating, but there is a conflict that cannot be resolved.

![2022-12-26_10-31](https://user-images.githubusercontent.com/16473797/209518633-05e56ea0-1695-43b4-9501-ea196f4e9673.png)

**Comment by Arsen6331** [comment]
While I think this should be solved in LURE (maybe by adding `--allowerasing` by default because that seems to be the behavior for all other distros), it is a good idea to provide a way to add flags to the package manager command, so I will definitely add that.

**Comment by Arsen6331** [comment]
I added `--allowerasing` in be79eba4c284d9e51dac390faf7c7b4d8b03a710 and the `-P` flag for passing arguments in 770881bf6762f27b5e0ea098d3e4af3d0374c020. The flag should be used like so:

```bash
lure -P '--print' in itd
```

Can you check to make sure it works as you expect?

**Comment by xpamych** [comment]
I'll check it out today, the end of the year. There is no free time at all.

**Comment by xpamych** [comment]
I realized that I don't know how to install the git version of lure)

**Comment by Arsen6331** [comment]
Just clone this git repo and do `sudo make install`. If you need to uninstall, do `sudo make uninstall`

**Comment by xpamych** [comment]
Hm) I have a problem))

main.go:38: running "scripts/gen-version.sh": exit status 128
internal/api/gen.go:3: running "protoc": exec: "protoc": executable file not found in $PATH
make: *** [Makefile:18: version.txt] Ошибка 1

**Comment by Arsen6331** [comment]
For now, just do `echo test > ./internal/config/version.txt`, `go build`, and then `sudo make install` again, there might be something wrong with one of my scripts.

**Comment by xpamych** [comment]
same error
![image](https://user-images.githubusercontent.com/16473797/210091261-41e3c2ee-adf9-40ef-91f7-e5c613923d6a.png)

**Comment by Arsen6331** [comment]
Ok, then try bypassing the makefile entirely and just doing `sudo install -Dm755 lure /usr/bin`

**Comment by xpamych** [comment]
```
lure -P '--allowerasing' in noisetorch                         ✔ 
--> Pulling repository name="default"
Enumerating objects: 518, done.
Counting objects: 100% (191/191), done.
Compressing objects: 100% (106/106), done.
Total 518 (delta 72), reused 141 (delta 43), pack-reused 327
 -> LURE repo's minumum LURE version is greater than the current version. Try updating LURE if something doesn't work. repo="default"
--> Pulling repository name="xpamych-gitflic"
Counting objects: 136, done
Finding sources: 100% (136/136)   
Getting sizes: 100% (39/39)   
Compressing objects: 100% (1673/1673)   
Total 1848 (delta 63), reused 1844 (delta 62)
panic: reflect.Set: value of type string is not assignable to type []string

goroutine 1 [running]:
reflect.Value.assignTo({0xe71c00?, 0xc00007c7a0?, 0x0?}, {0xfa7ea1, 0xb}, 0xe67ec0, 0x0)
	/usr/lib/golang/src/reflect/value.go:3064 +0x2ac
reflect.Value.Set({0xe67ec0?, 0xc0006a1428?, 0xf9b176?}, {0xe71c00?, 0xc00007c7a0?, 0xebf0c0?})
	/usr/lib/golang/src/reflect/value.go:2090 +0xeb
go.arsenm.dev/lure/internal/shutils/decoder.(*Decoder).DecodeVar.func1({0xe71c00?, 0xc00007c7a0?, 0x10?}, {0xf24780?, 0xc0006a1428?, 0x0?})
	/home/xpamych/git/lure/internal/shutils/decoder/decoder.go:73 +0xd1
github.com/mitchellh/mapstructure.DecodeHookExec({0xebf0c0?, 0x11b8670?}, {0xe71c00?, 0xc00007c7a0?, 0x5?}, {0xf24780?, 0xc0006a1428?, 0x203000?})
	/home/xpamych/go/pkg/mod/github.com/mitchellh/mapstructure@v1.5.0/decode_hooks.go:51 +0xd3
github.com/mitchellh/mapstructure.(*Decoder).decode(0xc000ba4e10, {0x0, 0x0}, {0xe71c00?, 0xc00007c7a0?}, {0xf24780?, 0xc0006a1428?, 0x10?})
	/home/xpamych/go/pkg/mod/github.com/mitchellh/mapstructure@v1.5.0/mapstructure.go:459 +0x179
github.com/mitchellh/mapstructure.(*Decoder).Decode(0xc000ba4e10, {0xe71c00, 0xc00007c7a0})
	/home/xpamych/go/pkg/mod/github.com/mitchellh/mapstructure@v1.5.0/mapstructure.go:417 +0xd8
go.arsenm.dev/lure/internal/shutils/decoder.(*Decoder).DecodeVar(0xef64a0?, {0xe4c4fb, 0xd}, {0xef64a0?, 0xc0006a1428})
	/home/xpamych/git/lure/internal/shutils/decoder/decoder.go:91 +0x10a
go.arsenm.dev/lure/internal/shutils/decoder.(*Decoder).DecodeVars(0xc000a94780?, {0xe57f00?, 0xc000a8de00})
	/home/xpamych/git/lure/internal/shutils/decoder/decoder.go:137 +0x4af
go.arsenm.dev/lure/internal/repos.parseScript({0x1316930, 0xc00094e9c0}, 0x0?, 0xc000a94780, {0x1313418?, 0xc000ba4c10?}, 0xc000b003f0?)
	/home/xpamych/git/lure/internal/repos/pull.go:378 +0x1a5
go.arsenm.dev/lure/internal/repos.processRepoFull({0x1316930, 0xc00094e9c0}, {{0xc00094c5a0?, 0x2e?}, {0xc0006bca00?, 0x0?}}, {0xc000039b30, 0x2e}, 0x1316930?)
	/home/xpamych/git/lure/internal/repos/pull.go:346 +0x65f
go.arsenm.dev/lure/internal/repos.Pull({0x1316930, 0xc00094e9c0}, 0x0?, {0xc00094e8c0?, 0x2, 0xc00058faa0?})
	/home/xpamych/git/lure/internal/repos/pull.go:118 +0x6d8
main.installCmd(0xc00052e2c0)
	/home/xpamych/git/lure/install.go:46 +0x1ac
github.com/urfave/cli/v2.(*Command).Run(0xc000518000, 0xc00052e1c0)
	/home/xpamych/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/command.go:173 +0x6a2
github.com/urfave/cli/v2.(*App).RunContext(0xc0001b4380, {0x1316930?, 0xc00094e9c0}, {0xc0000320a0, 0x5, 0x5})
	/home/xpamych/go/pkg/mod/github.com/urfave/cli/v2@v2.16.3/app.go:383 +0xfb5
main.main()
	/home/xpamych/git/lure/main.go:189 +0x11f3
```

**Comment by xpamych** [comment]
may be my distro borked?

**Comment by Arsen6331** [comment]
No, it looks like an issue with LURE's decoder, I'll have to look at that.

**Comment by Arsen6331** [comment]
I found the issue. LURE was not returning an error if the type of one of the variables was invalid, so it was panicking. The latest commit should return an error instead, which should look something like `variable 'architectures' is of type string, but []string is expected`, meaning the `architectures` variable is a string when an array was expected.

I'll close this now, since the feature this issue is for has been added.
